device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttMessaging.java

```diff
@@ -152,18 +154,18 @@ private boolean appendPropertyIfPresent(StringBuilder stringBuilder, boolean sep
                     stringBuilder.append(MESSAGE_PROPERTY_SEPARATOR);
                 }
 
-                stringBuilder.append(propertyKey);
+                stringBuilder.append(URIUtil.encodePath(propertyKey, StandardCharsets.UTF_8.name()));
```

Comment:
We weren't encoding the application property keys, which could be an issue. If a customer gives us a key with odd characters in it, we need to encode in order to keep the mqtt topic string valid